src/radical/saga/adaptors/shell/shell_job.py

```diff
@@ -622,6 +655,19 @@ def initialize(self):
         self._logger.debug("got mon prompt(%s)(%s)" % (ret, out.strip()))
 
 
+    # ----------------------------------------------------------------
+    #
+    def _ping_cb (self) :
+
+        with self._shell_lock:
+
+            _, out, _ = self.shell.run_sync('PING')
```

Comment:
it happens while `self.shell` is not created yet, but this callback is called (from travis):
```
Exception in thread Thread-50:
842Traceback (most recent call last):
843  File "/opt/python/3.6.7/lib/python3.6/threading.py", line 916, in _bootstrap_inner
844    self.run()
845  File "/opt/python/3.6.7/lib/python3.6/threading.py", line 1180, in run
846    self.finished.wait(self.interval)
847  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/radical/saga/adaptors/shell/shell_job.py", line 664, in _ping_cb
848    _, out, _ = self.shell.run_sync('PING')
849AttributeError: 'NoneType' object has no attribute 'run_sync'v
```
just add before this line `if self.shell:`?

Reply:
Then the test should call `init_instance` I guess?  SAGA has delayed instantiation for most classes, to support async instance creation.  So, unless `init_instance` is called, the class is not considered viable, and cannot really be tested either.

Reply:
yeah, that also works to move it, agree with that

Reply:
I read your comment incorrectly first... to call `init_instance` manually would help for test, but I didn't figure out how to put it there. But to check that `self.shell` doesn't contain None value shouldn't break anything

Reply:
> But to check that self.shell doesn't contain None value shouldn't break anything

well, no - but that implies that we would need to sprinkle such tests through the whole class to check for everything which is set up in the initializer, which is something which would obfuscate the code a lot, only to shield for missing initialization (and would then have to fail anyway).  So I'd rather not do that...

Reply:
That errors actually were after closing JobService... since Timer was still on. I've added `self._ping.cancel()` inside `close()` method

Reply:
Ah, that makes sense.  Thanks!